### PR TITLE
Convert all TODO(user) to TODO(issue)

### LINKED ISSUE #2450: All TODO should refer to GitHub issues
For now we use a lot of e.g. `TODO(michaelchirico)`, it's better to track an issue instead.

After fixing, remove user names as a valid option for `todo_regex_linter()` in our .lintr.

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #2450

Issues #2464 through #2493 are newly-formalized pending tasks spurred by this PR.

Some TODO were simply dropped (either I deemed them obsolete, or they were already fixed; I also dropped TODO in tests as not being the right place for them to live on as comments).

### CODE REVIEW

File: R/keyword_quote_linter.R [hunk 1]
```diff
@@ -46,11 +46,6 @@
 #' @evalRd rd_tags("keyword_quote_linter")
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
-# TODO(michaelchirico): offer a stricter version of this that
```
Comment (AshesITR) [inline R/keyword_quote_linter.R:49]: Issue reference lost

File: R/list_comparison_linter.R [hunk 2]
```diff
@@ -21,8 +21,6 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 list_comparison_linter <- function() {
-  # TODO(michaelchirico): extend to cases where using simplify=FALSE implies a
```
Comment (AshesITR) [inline R/list_comparison_linter.R:24]: ditto

File: R/literal_coercion_linter.R [hunk 3]
```diff
@@ -98,9 +98,6 @@ literal_coercion_linter <- function() {
         coercion_str[needs_prefix] <- paste0("rlang::", coercion_str[needs_prefix])
       }
       # the linter logic & rlang requirement should ensure that it's safe to run eval() here
-      # TODO(michaelchirico): this recommends '1' to replace as.numeric(1), where our
```
Comment (AshesITR) [inline R/literal_coercion_linter.R:101]: ditto

File: R/keyword_quote_linter.R (hunk 1 above)
Comment (MichaelChirico) [reply R/keyword_quote_linter.R:49]: This is #2471. I took a bit of a judgment call in some cases on totally removing the TODO from the sources, here's one of them. I think it's a bit odd to anchor this TODO to the function overall, interrupting the roxygen2 mark-up, so I removed it.

File: tests/testthat/test-any_duplicated_linter.R [hunk 4]
```diff
@@ -49,10 +49,6 @@ test_that("any_duplicated_linter catches length(unique()) equivalencies too", {
   expect_lint("length(unique(x)) != length(x)", lint_msg_x, linter)
   expect_lint("length(unique(x)) < length(x)", lint_msg_x, linter)
   expect_lint("length(x) > length(unique(x))", lint_msg_x, linter)
-
-  # TODO(michaelchirico): try and match data.table- and dplyr-specific versions of
```
Comment (AshesITR) [inline tests/testthat/test-any_duplicated_linter.R:53]: ditto

File: R/list_comparison_linter.R (hunk 2 above)
Comment (MichaelChirico) [reply R/list_comparison_linter.R:24]: Here I don't think the TODO anchors to this expression in particular, so I removed it. The issue is #2472.

File: R/literal_coercion_linter.R (hunk 3 above)
Comment (MichaelChirico) [reply R/literal_coercion_linter.R:101]: This one is a mistake, restoring.

File: tests/testthat/test-any_duplicated_linter.R (hunk 4 above)
Comment (MichaelChirico) [reply tests/testthat/test-any_duplicated_linter.R:53]: I removed all of the TODO from the tests, I don't think they should live there.

**Comment by AshesITR** [comment]
To help with reviewing, please comment the issue number in the PR where you removed the comment. Thanks for the house keeping!

File: R/is_numeric_linter.R [hunk 5]
```diff
@@ -34,11 +34,9 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 is_numeric_linter <- function() {
-  # TODO(michaelchirico): this should also cover is.double(x) || is.integer(x)
-  # TODO(#1636): is.numeric(x) || is.integer(x) || is.factor(x) is also redundant
-  # TODO(michaelchirico): consdier capturing any(class(x) == "numeric/integer")
```
Comment (MichaelChirico) [inline R/is_numeric_linter.R:39]: these two are combined in #2470

File: tests/testthat/test-expect_s3_class_linter.R [hunk 6]
```diff
@@ -60,14 +60,6 @@ test_that("expect_s3_class_linter blocks simple disallowed usages", {
     rex::rex("expect_s3_class(x, k) is better than expect_true(is.<k>(x))"),
     linter
   )
-
-  # TODO(michaelchirico): consider more carefully which sorts of class(x) %in% . and
```
Comment (MichaelChirico) [inline tests/testthat/test-expect_s3_class_linter.R:64]: #2482

File: tests/testthat/test-fixed_regex_linter.R [hunk 7]
```diff
@@ -270,19 +270,6 @@ test_that("fixed replacement is correct with UTF-8", {
   )
 })
 
-# TODO(michaelchirico): one difference for stringr functions vs. base is that
```
Comment (MichaelChirico) [inline tests/testthat/test-fixed_regex_linter.R:273]: This is done already, I think on the last release when we did pipe-awareness consistency.

File: tests/testthat/test-fixed_regex_linter.R [hunk 8]
```diff
@@ -270,19 +270,6 @@ test_that("fixed replacement is correct with UTF-8", {
...
-#   stringr is much friendlier to piping, so that
-#   > str %>% str_replace_all("x$", "y")
-#   actually doesn't need fixed(), but the logic now is only looking at "y"
-#   since it's the second argument and a non-regex string. Similarly,
-#   > str %>% str_detect("x")
-#   is a false negative. thankfully there appear to be few false positives here
-
-# TODO(michaelchirico): we could in principle build in logic to detect whether
```
Comment (MichaelChirico) [inline tests/testthat/test-fixed_regex_linter.R:281]: #2483

File: tests/testthat/test-if_not_else_linter.R [hunk 9]
```diff
@@ -83,5 +83,3 @@ test_that("exceptions= argument works", {
 
   expect_lint("if (!foo(x)) y else z", NULL, if_not_else_linter(exceptions = "foo"))
 })
-
-# TODO(michaelchirico): should if (A != B) be considered as well?
```
Comment (MichaelChirico) [inline tests/testthat/test-if_not_else_linter.R:87]: #2485

File: tests/testthat/test-ifelse_censor_linter.R [hunk 10]
```diff
@@ -81,6 +81,3 @@ test_that("lints vectorize", {
     ifelse_censor_linter()
   )
 })
-
-# TODO(michaelchirico): how easy would it be to strip parens when considering lint?
```
Comment (MichaelChirico) [inline tests/testthat/test-ifelse_censor_linter.R:85]: #2484

File: tests/testthat/test-knitr_formats.R [hunk 11]
```diff
@@ -94,18 +94,12 @@ test_that("it handles tex", {
     file = test_path("knitr_formats", "test.Rtex"),
     checks = list(
       list(regexes[["indent"]], line_number = 11L),
-      # TODO(AshesITR):
```
Comment (MichaelChirico) [inline tests/testthat/test-knitr_formats.R:97]: Both are #1043

File: tests/testthat/test-nzchar_linter.R [hunk 12]
```diff
@@ -17,9 +17,6 @@ test_that("nzchar_linter skips as appropriate for other nchar args", {
   # type="bytes" should be >= the value for the default (type="chars")
   expect_lint("nchar(x, type='width') == 0L", NULL, linter)
 
-  # TODO(michaelchirico): check compatibility of nchar(., allowNA=TRUE).
```
Comment (MichaelChirico) [inline tests/testthat/test-nzchar_linter.R:20]: #2487

File: tests/testthat/test-one_call_pipe_linter.R [hunk 13]
```diff
@@ -21,9 +21,7 @@ test_that("one_call_pipe_linter blocks simple disallowed usages", {
   # new lines don't matter
   expect_lint("x %>%\n  foo()", lint_msg, linter)
 
-  # catch the "inner" pipe chain, not the "outer" one
```
Comment (MichaelChirico) [inline tests/testthat/test-one_call_pipe_linter.R:24]: #2488

File: tests/testthat/test-unnecessary_lambda_linter.R [hunk 14]
```diff
@@ -167,8 +167,6 @@ test_that("unnecessary_lambda_linter doesn't apply to keyword args", {
 test_that("purrr-style anonymous functions are also caught", {
   linter <- unnecessary_lambda_linter()
 
-  # TODO(michaelchirico): this is just purrr::flatten(x). We should write another
```
Comment (MichaelChirico) [inline tests/testthat/test-unnecessary_lambda_linter.R:170]: #2489

File: tests/testthat/test-unnecessary_nesting_linter.R [hunk 15]
```diff
@@ -27,9 +27,6 @@ test_that("unnecessary_nesting_linter skips allowed usages", {
   )
 })
 
-# TODO(michaelchirico): consider if there's a nice easy pattern to enforce for
```
Comment (MichaelChirico) [inline tests/testthat/test-unnecessary_nesting_linter.R:30]: #2490

File: tests/testthat/test-unreachable_code_linter.R [hunk 16]
```diff
@@ -720,30 +720,3 @@ test_that("allow_comment_regex= obeys covr's custom exclusion when set", {
     linter_covr
   )
 })
-
-# nolint start: commented_code_linter.
-# TODO(michaelchirico): extend to work on switch() statements
```
Comment (MichaelChirico) [inline tests/testthat/test-unreachable_code_linter.R:725]: #2491

File: tests/testthat/test-unreachable_code_linter.R [hunk 17]
```diff
@@ -720,30 +720,3 @@ test_that("allow_comment_regex= obeys covr's custom exclusion when set", {
...
-#   ")
-#   expect_lint(
-#     unreachable_inside_switch_lines,
-#     rex::rex("Code and comments coming after a return() or stop()"),
-#     unreachable_code_linter()
-#   )
-# })
-# nolint end: commented_code_linter.
-
-# TODO(michaelchirico): This could also apply to cases without
```
Comment (MichaelChirico) [inline tests/testthat/test-unreachable_code_linter.R:748]: #2492

File: tests/testthat/test-yoda_test_linter.R [hunk 18]
```diff
@@ -61,15 +61,3 @@ test_that("lints vectorize", {
     yoda_test_linter()
   )
 })
-
-# TODO(michaelchirico): Should this be extended to RUnit tests? It seems yes,
```
Comment (MichaelChirico) [inline tests/testthat/test-yoda_test_linter.R:65]: Dropped this

File: tests/testthat/test-yoda_test_linter.R [hunk 19]
```diff
@@ -61,15 +61,3 @@ test_that("lints vectorize", {
...
-#   but the argument names in RUnit (inherited from base all.equal()) are a bit
-#   confusing, e.g. `checkEqual(target=, current=)`. From the name, one might
-#   reasonably conclude 'expected' comes first, and 'actual' comes second.
-# TODO(michaelchirico): What sorts of combinations of literals can be included?
```
Comment (MichaelChirico) [inline tests/testthat/test-yoda_test_linter.R:69]: #2493

File: tests/testthat/test-yoda_test_linter.R [hunk 20]
```diff
@@ -61,15 +61,3 @@ test_that("lints vectorize", {
...
-#   e.g. expect_equal(c(1, 2), x) is a yoda test; is expect_equal(c(x, 1), y)?
-#   clearly it's not true for general f() besides c(). What about other
-#   constructors of literals? data.frame(), data.table(), tibble(), ...?
-# TODO(michaelchirico): The logic could also be extended to "tests" inside regular
```
Comment (MichaelChirico) [inline tests/testthat/test-yoda_test_linter.R:73]: This is #1835

**Comment by MichaelChirico** [comment]
> To help with reviewing, please comment the issue number in the PR where you removed the comment. Thanks for the house keeping!

SG, done.

**Review by AshesITR — APPROVED** [review]
Thanks for the cleanup and your patience with this review :)

**Pull request merged**